src/main/java/de/qabel/desktop/ui/accounting/qrcode/QRCodeController.java

```diff
@@ -0,0 +1,114 @@
+package de.qabel.desktop.ui.accounting.qrcode;
+
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
+import de.qabel.core.config.Identity;
+import de.qabel.core.config.factory.DropUrlGenerator;
+import de.qabel.desktop.config.ClientConfig;
+import de.qabel.desktop.ui.AbstractController;
+import javafx.fxml.FXML;
+import javafx.fxml.Initializable;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.Pane;
+import net.glxn.qrgen.core.image.ImageType;
+import net.glxn.qrgen.javase.QRCode;
+
+import javax.inject.Inject;
+import java.io.ByteArrayInputStream;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.ResourceBundle;
+
+public class QRCodeController extends AbstractController implements Initializable {
+    @FXML
+    public Pane qrcode;
+
+    @FXML
+    private Label labelClient;
+
+    @FXML
+    private Label labelUrl;
+
+    @FXML
+    private Label labelKey;
+
+    @FXML
+    private ImageView imageQrCode;
+
+    @Inject
+    private Pane layoutWindow;
+
+    @Inject
+    private ClientConfig clientConfiguration;
+
+    @Inject
+    private DropUrlGenerator dropUrlGenerator;
+
+    @Inject
+    private URI accountingUri;
+
+    @Inject
+    private Identity identity;
+
+    private String dropUrl;
+
+    private String textQRCode;
+
+    @Override
+    public void initialize(URL location, ResourceBundle resources) {
+    }
+
+    public void buttonClosePopup() {
+        hidePopup();
+    }
+
+    public void showPopup() {
+        setIdentity();
+        qrcode.setVisible(true);
+    }
+
+    private void hidePopup() {
+        qrcode.setVisible(false);
+        layoutWindow.getChildren().remove(this);
```

Comment:
Don't think this actually works. `this` is the controller but the `view` needs to be removed.
As a result, the view will stay as an (invisible) child of the layout. You could remove the `pane` and add it again on `showPopup`.